File: decompiled-src-verify/io/legado/app/model/analyzeRule/AnalyzeByJSonPath.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  com.jayway.jsonpath.DocumentContext
 *  com.jayway.jsonpath.JsonPath
 *  com.jayway.jsonpath.Predicate
 *  com.jayway.jsonpath.ReadContext
 *  kotlin.Metadata
 *  kotlin.collections.CollectionsKt
 *  kotlin.jvm.functions.Function1
 *  kotlin.jvm.internal.DefaultConstructorMarker
 *  kotlin.jvm.internal.Intrinsics
 *  org.jetbrains.annotations.NotNull
 *  org.jetbrains.annotations.Nullable
 */
package io.legado.app.model.analyzeRule;

import com.jayway.jsonpath.DocumentContext;
import com.jayway.jsonpath.JsonPath;
import com.jayway.jsonpath.Predicate;
import com.jayway.jsonpath.ReadContext;
import io.legado.app.model.analyzeRule.RuleAnalyzer;
import java.util.ArrayList;
import java.util.List;
import kotlin.Metadata;
import kotlin.collections.CollectionsKt;
import kotlin.jvm.functions.Function1;
import kotlin.jvm.internal.DefaultConstructorMarker;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Metadata(mv={1, 5, 1}, k=1, xi=48, d1={"\u0000(\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0002\b\u0005\n\u0002\u0010 \n\u0002\b\u0003\u0018\u0000 \u00112\u00020\u0001:\u0001\u0011B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0001\u00a2\u0006\u0002\u0010\u0003J\u001d\u0010\u0006\u001a\n\u0012\u0004\u0012\u00020\u0001\u0018\u00010\u00072\u0006\u0010\b\u001a\u00020\tH\u0000\u00a2\u0006\u0002\b\nJ\u0015\u0010\u000b\u001a\u00020\u00012\u0006\u0010\b\u001a\u00020\tH\u0000\u00a2\u0006\u0002\b\fJ\u0010\u0010\r\u001a\u0004\u0018\u00010\t2\u0006\u0010\b\u001a\u00020\tJ\u001b\u0010\u000e\u001a\b\u0012\u0004\u0012\u00020\t0\u000f2\u0006\u0010\b\u001a\u00020\tH\u0000\u00a2\u0006\u0002\b\u0010R\u000e\u0010\u0004\u001a\u00020\u0005X\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0012"}, d2={"Lio/legado/app/model/analyzeRule/AnalyzeByJSonPath;", "", "json", "(Ljava/lang/Object;)V", "ctx", "Lcom/jayway/jsonpath/ReadContext;", "getList", "Ljava/util/ArrayList;", "rule", "", "getList$reader_pro", "getObject", "getObject$reader_pro", "getString", "getStringList", "", "getStringList$reader_pro", "Companion", "reader-pro"})
public final class AnalyzeByJSonPath {
    @NotNull
    public static final Companion Companion = new Companion(null);
    @NotNull
    private ReadContext ctx;

    public AnalyzeByJSonPath(@NotNull Object json) {
        Intrinsics.checkNotNullParameter((Object)json, (String)"json");
        this.ctx = Companion.parse(json);
    }

    @Nullable
    public final String getString(@NotNull String rule) {
        Intrinsics.checkNotNullParameter((Object)rule, (String)"rule");
        CharSequence charSequence = rule;
        boolean bl = false;
        if (charSequence.length() == 0) {
            return null;
        }
        String result2 = null;
        RuleAnalyzer ruleAnalyzes = new RuleAnalyzer(rule, true);
        Object object = new String[]{"&&", "||"};
        ArrayList<String> rules = ruleAnalyzes.splitRule((String[])object);
        if (rules.size() == 1) {
            ruleAnalyzes.reSetPos();
            result2 = RuleAnalyzer.innerRule$default(ruleAnalyzes, "{$.", 0, 0, (Function1)new Function1<String, String>(this){
                final /* synthetic */ AnalyzeByJSonPath this$0;
                {
                    this.this$0 = $receiver;
                    super(1);
                }

                @Nullable
                public final String invoke(@NotNull String it) {
                    Intrinsics.checkNotNullParameter((Object)it, (String)"it");
                    return this.this$0.getString(it);
                }
            }, 6, null);
            object = result2;
            boolean bl2 = false;
            if (object.length() == 0) {
                try {
                    Object ob = this.ctx.read(rule, new Predicate[0]);
                    result2 = ob instanceof List ? CollectionsKt.joinToString$default((Iterable)((Iterable)ob), (CharSequence)"\n", null, null, (int)0, null, null, (int)62, null) : ob.toString();
                }
                catch (Exception e) {
                    e.printStackTrace();
                }
            }
            return result2;
        }
        boolean bl3 = false;
        ArrayList<String> textList = new ArrayList<String>();
        for (String rl : rules) {
            Intrinsics.checkNotNullExpressionValue((Object)rl, (String)"rl");
            String temp = this.getString(rl);
            CharSequence charSequence2 = temp;
            boolean bl4 = false;
            boolean bl5 = false;
            if (charSequence2 == null || charSequence2.length() == 0) continue;
            textList.add(temp);
            if (!Intrinsics.areEqual((Object)ruleAnalyzes.getElementsType(), (Object)"||")) continue;
            break;
        }
        return CollectionsKt.joinToString$default((Iterable)textList, (CharSequence)"\n", null, null, (int)0, null, null, (int)62, null);
    }

    @NotNull
    public final List<String> getStringList$reader_pro(@NotNull String rule) {
        Intrinsics.checkNotNullParameter((Object)rule, (String)"rule");
        ArrayList<String> result2 = new ArrayList<String>();
        CharSequence charSequence = rule;
        boolean bl = false;
        if (charSequence.length() == 0) {
            return result2;
        }
        RuleAnalyzer ruleAnalyzes = new RuleAnalyzer(rule, true);
        String[] stringArray = new String[]{"&&", "||", "%%"};
        ArrayList<String> rules = ruleAnalyzes.splitRule(stringArray);
        if (rules.size() == 1) {
            block16: {
                ruleAnalyzes.reSetPos();
                String st2 = RuleAnalyzer.innerRule$default(ruleAnalyzes, "{$.", 0, 0, (Function1)new Function1<String, String>(this){
                    final /* synthetic */ AnalyzeByJSonPath this$0;
                    {
                        this.this$0 = $receiver;
                        super(1);
                    }

                    @Nullable
                    public final String invoke(@NotNull String it) {
                        Intrinsics.checkNotNullParameter((Object)it, (String)"it");
                        return this.this$0.getString(it);
                    }
                }, 6, null);
                CharSequence charSequence2 = st2;
                boolean bl2 = false;
                if (charSequence2.length() == 0) {
                    try {
                        Object obj = this.ctx.read(rule, new Predicate[0]);
                        if (obj instanceof List) {
                            for (Object o : (List)obj) {
                                result2.add(String.valueOf(o));
                            }
                            break block16;
                        }
                        result2.add(obj.toString());
                    }
                    catch (Exception e) {
                        e.printStackTrace();
                    }
                } else {
                    result2.add(st2);
                }
            }
            return result2;
        }
        ArrayList<List<String>> results = new ArrayList<List<String>>();
        for (String string : rules) {
            Intrinsics.checkNotNullExpressionValue((Object)string, (String)"rl");
            List<String> temp = this.getStringList$reader_pro(string);
            Object object = temp;
            boolean bl2 = false;
            if (!(!object.isEmpty())) continue;
            results.add(temp);
            object = temp;
            bl2 = false;
            if (!(!object.isEmpty()) || !Intrinsics.areEqual((Object)ruleAnalyzes.getElementsType(), (Object)"||")) continue;
        }
        if (results.size() > 0) {
            if (Intrinsics.areEqual((Object)"%%", (Object)ruleAnalyzes.getElementsType())) {
                int n = 0;
                int n2 = ((List)results.get(0)).size() + -1;
                if (n <= n2) {
                    do {
                        int i = n++;
                        for (List list2 : results) {
                            if (i >= list2.size()) continue;
                            result2.add((String)list2.get(i));
                        }
                    } while (n <= n2);
                }
            } else {
                for (List list3 : results) {
                    result2.addAll(list3);
                }
            }
        }
        return result2;
    }

    @NotNull
    public final Object getObject$reader_pro(@NotNull String rule) {
        Intrinsics.checkNotNullParameter((Object)rule, (String)"rule");
        Object object = this.ctx.read(rule, new Predicate[0]);
        Intrinsics.checkNotNullExpressionValue((Object)object, (String)"ctx.read(rule)");
        return object;
    }

    @Nullable
    public final ArrayList<Object> getList$reader_pro(@NotNull String rule) {
        Intrinsics.checkNotNullParameter((Object)rule, (String)"rule");
        ArrayList<Object> result2 = new ArrayList<Object>();
        CharSequence charSequence = rule;
        boolean bl = false;
        if (charSequence.length() == 0) {
            return result2;
        }
        RuleAnalyzer ruleAnalyzes = new RuleAnalyzer(rule, true);
        String[] stringArray = new String[]{"&&", "||", "%%"};
        ArrayList<String> rules = ruleAnalyzes.splitRule(stringArray);
        if (rules.size() == 1) {
            stringArray = this.ctx;
            boolean bl2 = false;
            boolean bl3 = false;
            String[] it = stringArray;
            boolean bl4 = false;
            try {
                return (ArrayList)it.read(rules.get(0), new Predicate[0]);
            }
            catch (Exception exception) {
                exception.printStackTrace();
            }
        } else {
            ArrayList<ArrayList<Object>> results = new ArrayList<ArrayList<Object>>();
            for (String string : rules) {
                Intrinsics.checkNotNullExpressionValue((Object)string, (String)"rl");
                ArrayList<Object> temp = this.getList$reader_pro(string);
                if (temp == null) continue;
                Object object = temp;
                boolean bl2 = false;
                if (!(!object.isEmpty())) continue;
                results.add(temp);
                object = temp;
                bl2 = false;
                if (!(!object.isEmpty()) || !Intrinsics.areEqual((Object)ruleAnalyzes.getElementsType(), (Object)"||")) continue;
            }
            if (results.size() > 0) {
                if (Intrinsics.areEqual((Object)"%%", (Object)ruleAnalyzes.getElementsType())) {
                    int n = 0;
                    int n2 = ((ArrayList)results.get(0)).size();
                    if (n < n2) {
                        do {
                            int i = n++;
                            for (ArrayList arrayList : results) {
                                Object e;
                                if (i >= arrayList.size() || (e = arrayList.get(i)) == null) continue;
                                Object e2 = e;
                                boolean bl3 = false;
                                boolean bl4 = false;
                                Object it = e2;
                                boolean bl7 = false;
                                result2.add(it);
                            }
                        } while (n < n2);
                    }
                } else {
                    for (ArrayList arrayList : results) {
                        result2.addAll(arrayList);
                    }
                }
            }
        }
        return result2;
    }

    @Metadata(mv={1, 5, 1}, k=1, xi=48, d1={"\u0000\u0014\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\b\u0086\u0003\u0018\u00002\u00020\u0001B\u0007\b\u0002\u00a2\u0006\u0002\u0010\u0002J\u000e\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u0001\u00a8\u0006\u0006"}, d2={"Lio/legado/app/model/analyzeRule/AnalyzeByJSonPath$Companion;", "", "()V", "parse", "Lcom/jayway/jsonpath/ReadContext;", "json", "reader-pro"})
    public static final class Companion {
        private Companion() {
        }

        @NotNull
        public final ReadContext parse(@NotNull Object json) {
            ReadContext readContext;
            Intrinsics.checkNotNullParameter((Object)json, (String)"json");
            Object object = json;
            if (object instanceof ReadContext) {
                readContext = (ReadContext)json;
            } else if (object instanceof String) {
                DocumentContext documentContext = JsonPath.parse((String)((String)json));
                Intrinsics.checkNotNullExpressionValue((Object)documentContext, (String)"parse(json)");
                readContext = (ReadContext)documentContext;
            } else {
                DocumentContext documentContext = JsonPath.parse((Object)json);
                Intrinsics.checkNotNullExpressionValue((Object)documentContext, (String)"parse(json)");
                readContext = (ReadContext)documentContext;
            }
            return readContext;
        }

        public /* synthetic */ Companion(DefaultConstructorMarker $constructor_marker) {
            this();
        }
    }
}

File: best-of-3/src/io/legado/app/model/analyzeRule/AnalyzeByJSonPath.kt
package io.legado.app.model.analyzeRule

import com.jayway.jsonpath.DocumentContext
import com.jayway.jsonpath.JsonPath
import com.jayway.jsonpath.ReadContext
import java.util.ArrayList
import kotlin.jvm.functions.Function1
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

public class AnalyzeByJSonPath(json: Any) {
   private final var ctx: ReadContext

   init {
      this.ctx = Companion.parse(json);
   }

   public fun getString(rule: String): String? {
      if (rule.length() == 0) {
         return null;
      } else {
         val var15: RuleAnalyzer = new RuleAnalyzer(rule, true);
         val rules: ArrayList = var15.splitRule("&&", "||");
         if (rules.size() == 1) {
            var15.reSetPos();
            var var14: java.lang.String = RuleAnalyzer.innerRule$default(var15, "{$.", 0, 0, (new Function1<java.lang.String, java.lang.String>(this) {
               {
                  super(1);
                  this.this$0 = `$receiver`;
               }

               @Nullable
               public final java.lang.String invoke(@NotNull java.lang.String it) {
                  return this.this$0.getString(it);
               }
            }) as Function1, 6, null);
            if (var14.length() == 0) {
               try {
                  val var18: Any = this.ctx.read(rule);
                  var14 = if (var18 is java.util.List)
                     CollectionsKt.joinToString$default(var18 as java.lang.Iterable, "\n", null, null, 0, null, null, 62, null)
                     else
                     var18.toString();
               } catch (var12: Exception) {
                  var12.printStackTrace();
               }
            }

            return var14;
         } else {
            val var16: ArrayList = new ArrayList();

            for (java.lang.String rl : rules) {
               val temp: java.lang.String = this.getString(rl);
               if (temp != null && temp.length() != 0) {
                  var16.add(temp);
                  if (var15.getElementsType() == "||") {
                     break;
                  }
               }
            }

            return CollectionsKt.joinToString$default(var16, "\n", null, null, 0, null, null, 62, null);
         }
      }
   }

   internal fun getStringList(rule: String): List<String> {
      val result: ArrayList = new ArrayList();
      if (rule.length() == 0) {
         return result;
      } else {
         val var12: RuleAnalyzer = new RuleAnalyzer(rule, true);
         val var13: ArrayList = var12.splitRule("&&", "||", "%%");
         if (var13.size() == 1) {
            var12.reSetPos();
            val var15: java.lang.String = RuleAnalyzer.innerRule$default(var12, "{$.", 0, 0, (new Function1<java.lang.String, java.lang.String>(this) {
               {
                  super(1);
                  this.this$0 = `$receiver`;
               }

               @Nullable
               public final java.lang.String invoke(@NotNull java.lang.String it) {
                  return this.this$0.getString(it);
               }
            }) as Function1, 6, null);
            if (var15.length() == 0) {
               try {
                  val var19: Any = this.ctx.read(rule);
                  if (var19 is java.util.List) {
                     for (Object o : (java.util.List)obj) {
                        result.add(java.lang.String.valueOf(var25));
                     }
                  } else {
                     result.add(var19.toString());
                  }
               } catch (var11: Exception) {
                  var11.printStackTrace();
               }
            } else {
               result.add(var15);
            }

            return result;
         } else {
            val var14: ArrayList = new ArrayList();

            for (java.lang.String rl : rules) {
               val i: java.util.List = this.getStringList$reader_pro(temp);
               if (!i.isEmpty()) {
                  var14.add(i);
                  if (!i.isEmpty() && var12.getElementsType() == "||") {
                     break;
                  }
               }
            }

            if (var14.size() > 0) {
               if ("%%" == var12.getElementsType()) {
                  var var16: Int = 0;
                  val var20: Int = (var14.get(0) as java.util.List).size() + -1;
                  if (0 <= var20) {
                     do {
                        val var24: Int = var16++;

                        for (java.util.List temp : results) {
                           if (var24 < var29.size()) {
                              result.add(var29.get(var24));
                           }
                        }
                     } while (var16 <= var20);
                  }
               } else {
                  for (java.util.List tempx : results) {
                     result.addAll(tempx);
                  }
               }
            }

            return result;
         }
      }
   }

   internal fun getObject(rule: String): Any {
      val var2: Any = this.ctx.read(rule);
      return var2;
   }

   internal fun getList(rule: String): ArrayList<Any>? {
      val result: ArrayList = new ArrayList();
      if (rule.length() == 0) {
         return result;
      } else {
         val var18: RuleAnalyzer = new RuleAnalyzer(rule, true);
         val var19: ArrayList = var18.splitRule("&&", "||", "%%");
         if (var19.size() == 1) {
            val i: ReadContext = this.ctx;

            try {
               return i.read(var19.get(0) as java.lang.String);
            } catch (var17: Exception) {
               var17.printStackTrace();
            }
         } else {
            val var21: ArrayList = new ArrayList();

            for (java.lang.String rl : rules) {
               val var28: ArrayList = this.getList$reader_pro(var25);
               if (var28 != null && !var28.isEmpty()) {
                  var21.add(var28);
                  if (!var28.isEmpty() && var18.getElementsType() == "||") {
                     break;
                  }
               }
            }

            if (var21.size() > 0) {
               if ("%%" == var18.getElementsType()) {
                  var var23: Int = 0;
                  val var26: Int = (var21.get(0) as ArrayList).size();
                  if (0 < var26) {
                     do {
                        val var29: Int = var23++;

                        for (ArrayList temp : results) {
                           if (var29 < var34.size()) {
                              val var11: Any = var34.get(var29);
                              if (var11 != null) {
                                 result.add(var11);
                              }
                           }
                        }
                     } while (var23 < var26);
                  }
               } else {
                  for (ArrayList tempx : results) {
                     result.addAll(tempx);
                  }
               }
            }
         }

         return result;
      }
   }

   public companion object {
      public fun parse(json: Any): ReadContext {
         val var10000: ReadContext;
         if (json is ReadContext) {
            var10000 = json as ReadContext;
         } else if (json is java.lang.String) {
            val var3: DocumentContext = JsonPath.parse(json as java.lang.String);
            var10000 = var3;
         } else {
            val var4: DocumentContext = JsonPath.parse(json);
            var10000 = var4;
         }

         return var10000;
      }
   }
}
